### PR TITLE
Update Build, Replace Bower

### PR DESCRIPTION
**Pull request by johnantoni**
What's included:

* Updated grunt packages, added node-sass as the SCSS compiler for the sass grunt task.
* Remove Bower, as no longer maintained.
* Update smoothScroll to latest and navBar.js to work with the updated library, naming changed from smoothScroll() to SmoothScroll().
* Update readme.md to remove reference to Bower, as no longer needed.

Ready for review.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -1,7 +1,23 @@
 'use strict';
 
+const sass = require('node-sass');
```
Comment (zombieFox) [inline Gruntfile.js]: Again, just for my education. What is the reason behind using `node-sass` and setting it to a `const`?

File: Gruntfile.js [hunk 2]
```diff
@@ -1,7 +1,23 @@
...
+
 module.exports = function(grunt) {
 
+  grunt.loadNpmTasks('grunt-contrib-htmlmin');
+  grunt.loadNpmTasks('grunt-sass');
```
Comment (zombieFox) [inline Gruntfile.js]: Is `grunt-sass` still needed if `node-sass` is used above?

File: Gruntfile.js [hunk 3]
```diff
@@ -146,6 +164,7 @@ module.exports = function(grunt) {
     sass: {
       dev: {
         options: {
+          implementation: sass,
```
Comment (zombieFox) [inline Gruntfile.js]: I'm so in the dark, what does `implementation` mean?

File: Gruntfile.js [hunk 4]
```diff
@@ -313,28 +333,14 @@ module.exports = function(grunt) {
...
-  grunt.loadNpmTasks('grunt-contrib-concat');
-  grunt.loadNpmTasks('grunt-contrib-cssmin');
-  grunt.loadNpmTasks('grunt-contrib-copy');
-  grunt.loadNpmTasks('grunt-contrib-uglify');
-  grunt.loadNpmTasks('grunt-contrib-connect');
-  grunt.loadNpmTasks('grunt-usemin');
-  grunt.loadNpmTasks('grunt-contrib-clean');
-  grunt.loadNpmTasks('grunt-assemble');
-  grunt.loadNpmTasks('grunt-autoprefixer');
-  grunt.loadNpmTasks('grunt-sw-precache');
```
Comment (zombieFox) [inline Gruntfile.js]: Is it better practice to keep the `loadNpmTasks` at the top of the file?

File: build/css/awesomeSheet.css [hunk 5]
```diff
@@ -1364,22 +1363,17 @@ input[type="submit"] {
...
-      -ms-flex-align: center;
-          align-items: center; }
+  -ms-flex-flow: row nowrap;
+      flex-flow: row nowrap;
+  -ms-flex-pack: center;
+      justify-content: center;
+  -ms-flex-align: center;
+      align-items: center; }
   .button-icon span {
-    -webkit-flex: 0 0 auto;
```
Comment (zombieFox) [inline build/css/awesomeSheet.css]: Has the `autoprefixer` changed? I would like to keep the `-ms-*` vendor prefixes for extra safety. 
I wasn't about to track down what was causing this.

Comment (zombieFox) [reply build/css/awesomeSheet.css]: Also, I've had issues in the past with different versions of Safari not recognising flexbox without the `-webkit-*` prefix.

File: Gruntfile.js (hunk 1 above)
Comment (johnantoni) [reply Gruntfile.js]: we can use a var or let but as it's just a reference to node-sass and it won't be altered along the way it's better to have it as a const.

File: Gruntfile.js (hunk 3 above)
Comment (johnantoni) [reply Gruntfile.js]: that's a new parameter needed for the grunt-sass library, https://github.com/sindresorhus/grunt-sass

File: Gruntfile.js (hunk 4 above)
Comment (johnantoni) [reply Gruntfile.js]: personal preference, i can move them back to the bottom but i like to have them up the top so you can see what kind of things you'll be using beforehand.

File: build/css/awesomeSheet.css (hunk 5 above)
Comment (johnantoni) [reply build/css/awesomeSheet.css]: this one's a weird one, as awesomeSheet.css it built from the grunt build job i reckon the sass compiler determined -webkit-flex was no longer needed in the compile https://caniuse.com/#feat=flexbox which sounds about right. I think generally we should ignore the /build directory in the .gitignore, what do you think?

Comment (carloscasalar) [reply build/css/awesomeSheet.css]: Sure, you should ignore the build files.

**Comment by johnantoni** [comment]
removed the /build directory from the git repo

**Comment by johnantoni** [comment]
@zombieFox good to merge in?

**Comment by zombieFox** [comment]
Apologies, I've had a busy couple of weeks (family in town). I've spotted a few issues while testing last week and have been looking to resolve them. These include:

- Smooth scrolling no long working with the `clone.js` module
- Smooth scrolling incorrectly activating the wrong icons in the nav

I've got some fixes in place now. I'll attempt to add a commit to this PR.

File: build/css/awesomeSheet.css (hunk 5 above)
Comment (zombieFox) [reply build/css/awesomeSheet.css]: The `/build/` directory is needed  as that is pushed to the `gh-pages` branch.

**Comment by zombieFox** [comment]
Good to merge once the `/build/` removal change is resolved.

**Comment by johnantoni** [comment]
Ok, the build dir should be ignored now on commits. Had to add the dir, remove it, then add it it gitignore commit and then build it again. after that looks to be no longer getting committed.

**Comment by zombieFox** [comment]
I'm still not sure how the build and deploy will work without a `/build/` directory, I'm better you have a plan. 😄

**Pull request merged**